### ISSUE TITLE
hr_employee_id Error

### ISSUE THREAD

**Issue by mnaphade** [OCA/hr#520]
I have hr_employee_id module installed on 11.0-20181121 CE. Everything seems alright for the privileged users. However, when employees login with their user credentials, they see an error upon clicking "Employees" module in their login.
```
Error:
Uncaught Error: QWeb2 - template['kanban-box']: Runtime Error: TypeError: Cannot read property 'raw_value' of undefined

https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3689
Traceback:
Error: QWeb2 - template['kanban-box']: Runtime Error: TypeError: Cannot read property 'raw_value' of undefined
    at Object.exception (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3689:7)
    at Engine.eval (eval at _render (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3731:73), <anonymous>:73:29)
    at Engine._render (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3730:296)
    at Engine.render (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3730:151)
    at Engine._render (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3734:57)
    at Engine.render (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3730:151)
    at Class._render (https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:1534:118)
    at https://portal.company.com/web/content/509-691ab44/web.assets_common.js:802:681
    at fire (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:796:299)
    at Object.add [as done] (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:797:467)
```

If I uninstall hr_employee_id module everything seems to work alright. Any ideas?

**Comment by pedrobaeza** [comment]
Hi, thanks for reporting. Are you able to reproduce the problem in runbot infrastructure (https://runbot.odoo-community.org/runbot/116/11.0 - admin/admin)?

If so, please explain with all the possible detailed steps to get to the error. Not having them, it's very difficult to hunt down the problem. Please don't be shy of adding details, and even better if you provide some screenshots or videos.

If not, this can be due to a configuration change, a custom/community module you have deployed in your DB or an outdated code. Please report back if it's the case.

For keeping a clean house in this issue tracker, if you don't answer in several days, this will be closed assuming it was one of the stated things.

**Comment by mnaphade** [comment]
Hello @pedrobaeza 
Thank you for replying. There aren't a whole lot of modules installed apart from the basic ones that ship with the installer of v11 CE. Attached is the list of modules that are currently installed on the system.
It appears that the issue is reproducible on runbot too. Here are the steps to reproduce

1. No changes to the default values of hr_employee_id in the admin/admin backend
2. Logon to the runbot using demo/demo
3. Click on Employees, and following error shows up there too. Similar to my installation.

```
Error:
Uncaught Error: QWeb2 - template['kanban-box']: Runtime Error: TypeError: Cannot read property 'raw_value' of undefined

http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:3689
Traceback:
Error: QWeb2 - template['kanban-box']: Runtime Error: TypeError: Cannot read property 'raw_value' of undefined
    at Object.exception (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:3689:7)
    at Engine.eval (eval at _render (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:3731:73), <anonymous>:73:29)
    at Engine._render (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:3730:296)
    at Engine.render (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:3730:151)
    at Engine._render (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:3734:57)
    at Engine.render (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:3730:151)
    at Class._render (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/574-68d2af5/web.assets_backend.js:1534:118)
    at http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:802:681
    at fire (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:796:299)
    at Object.add [as done] (http://3347330-11-0-e1dea8.runbot1.odoo-community.org/web/content/560-7f1f9a8/web.assets_common.js:797:467)
```

[Odoo Installed Modules.xlsx](https://github.com/OCA/hr/files/2637135/Odoo.Installed.Modules.xlsx)

**Comment by pedrobaeza** [comment]
@Trivedi-Vacha-SerpentCS, you were the porter of this. Can you please check?

**Comment by mnaphade** [comment]
A quick update:

I edited the "HR - Employees Kanban" in the views and removed .raw_value from the view. Now the edited view looks something like below:

```
<?xml version="1.0"?>
<li id="last_login" position="after">
                <li t-if="record.identification_id">
                    <field name="identification_id"/>
                </li>
            </li>
```

With this, I can now at least click on the "Employees" menu from employee login. I can now see the employees however when I click the employee records, I see below error:

Error:
Uncaught TypeError: Cannot read property 'string' of undefined

```
https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:1400
Traceback:
TypeError: Cannot read property 'string' of undefined
    at Class._renderTagLabel (https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:1400:868)
    at Class._renderNode [as _super] (https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:1390:246)
    at Class._renderNode (https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:2339:51)
    at Class._renderNode (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3819:371)
    at Function._.map._.collect (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:13:270)
    at Class._renderGenericTag (https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:1382:303)
    at Class._renderNode [as _super] (https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:1393:13)
    at Class._renderNode (https://portal.company.com/web/content/516-50c604e/web.assets_backend.js:2339:51)
    at Class._renderNode (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:3819:371)
    at Function._.map._.collect (https://portal.company.com/web/content/509-691ab44/web.assets_common.js:13:270)
```

**Comment by Trivedi-Vacha-SerpentCS** [comment]
> @Trivedi-Vacha-SerpentCS, you were the porter of this. Can you please check?

@pedrobaeza Sure

**Comment by Trivedi-Vacha-SerpentCS** [comment]
Fixed in pr #521

**Comment by alexey-pelykh** [comment]
In 12.0 it was addressed a bit differently, and it would be great to sync up on approach on fix https://github.com/OCA/hr/blob/12.0/hr_employee_id/views/hr_employee_views.xml
